src/coreclr/nativeaot/System.Private.CoreLib/src/System/GC.NativeAot.cs

```diff
@@ -22,7 +22,8 @@ public enum GCCollectionMode
     {
         Default = 0,
         Forced = 1,
-        Optimized = 2
+        Optimized = 2,
+        Aggressive = 3,
```

Comment:
maybe it's a bit better to add a 

`MaxCollectionMode = Aggressive` 

here so every time we add another mode we don't have to update our checks below like 

`if ((mode < GCCollectionMode.Default) || (mode > GCCollectionMode.Aggressive))`
? 

just do 

`if ((mode < GCCollectionMode.Default) || (mode > GCCollectionMode.MaxCollectionMode))`

instead

Reply:
That's against [API guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum) because we generally can't update enum fields (that's a breaking change). Code that was compiled with a check like `if ((mode < GCCollectionMode.Default) || (mode >= GCCollectionMode.MaxCollectionMode))` would be broken unless it happens to be in the same module as `GCCollectionMode`.

![image](https://user-images.githubusercontent.com/5169960/170783264-c5ed1e6c-3492-4d3a-ba9a-7d5a49044574.png)